packages/devtools/lib/src/memory/memory_plotly.dart

```diff
@@ -0,0 +1,192 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools/src/memory/memory.dart';
+import '../ui/plotly.dart';
+
+class MemoryPlotly {
+  MemoryPlotly(this._domName, this._memoryChart);
+
+  final String _domName;
+  final MemoryChart _memoryChart;
+
+  Layout getMemoryLayout(chartTitle) {
+    return Layout(
+        title: chartTitle,
+        xaxis: AxisLayout(
+          type: 'date',
+          tickformat: '%-I:%M:%S %p',
+          hoverformat: '%M:%S.%L %p',
+          range: [],
+          rangeslider: RangeSlider(),
+        ),
+        yaxis: AxisLayout(
+          title: 'Heap',
+          fixedrange: true,
+        ),
+        margin: Margin(l: 80, r: 0, b: 5, t: 5, pad: 5));
+  }
+
+//  static const int MEMORY_GC_TRACE = 0;   // TODO(terry):
+  static const int MEMORY_EXTERNAL_TRACE = 0;
+  static const int MEMORY_USED_TRACE = 1;
+  static const int MEMORY_CAPACITY_TRACE = 2;
+  static const int MEMORY_RSS_TRACE = 3;
+
+  // TODO(terry): expose as CSSColor and Themed.
+  static const String RSS_COLOR = '#F1B876';
+  static const String CAPACITY_COLOR = '#C0C0C0';
+  static const String EXTERNAL_COLOR = '#4794C0';
+  static const String USED_COLOR = '#48B2E7';
+
+  List<Data> createMemoryTraces() {
+    /* TODO(terry): Enable gc markers.
+    final Data normalized_trace = Data(
+      y: [],
+      x: [],
+      type: 'scatter',
+      mode: 'markers',
+      marker: Marker(
+        symbol: 'circle',
+        size: 10,
+      ),
+      name: 'GC',
+      text: [],
+      hoverinfo: 'y+name',
+    );
+    */
+
+    final Data externalTrace = Data(
```

Comment:
add a helper method so it is easier to see what is different about each one of these Data objects. Most look the same except for the group, name, and color.

Reply:
Done.